ui/components/app/wallet-overview/coin-buttons.tsx

```diff
@@ -356,15 +368,18 @@ const CoinButtons = ({
     );
     switch (account.type) {
```

Comment:
Similarly to the refactor of the `BtcOverview` component to `NonEvmOverview`, maybe we could use another pattern here like:

```suggestion
    if (isEvmAccountType(account.type)) {
      trackEvent(
        {
          event: MetaMetricsEventName.NavSendButtonClicked,
          category: MetaMetricsEventCategory.Navigation,
          properties: {
            token_symbol: 'ETH',
            location: 'Home',
            text: 'Send',
            chain_id: chainId,
          },
        },
        { excludeMetaMetricsId: false },
      );
      await setCorrectChain();
      await dispatch(startNewDraftTransaction({ type: AssetType.native }));
      history.push(SEND_ROUTE);
    } else {
      if (!account.metadata.snap) {
        throw new Error('Non-EVM account needs to be Snap accounts');
      }
    
      try {
        // FIXME: We switch the tab before starting the send flow (we
        // faced some inconsistencies when changing it after).
        await dispatch(setDefaultHomeActiveTabName('activity'));
        await sendMultichainTransaction(
          account.metadata.snap.id,
          {
            account: account.id,
            scope: chainId as CaipChainId,
          },
        );
      } catch {
        // Restore the previous tab in case of any error (see FIXME comment above).
        await dispatch(setDefaultHomeActiveTabName(currentActivityTabName));
      }
    }
```